gradle-plugin/src/main/java/com/contrastsecurity/gradle/plugin/ContrastVerifyTestTask.java

```diff
@@ -1,15 +1,229 @@
+   */
+  private List<String> getServerId(
+      final ContrastSDK sdk,
+      final String applicationId,
+      final String orgUuid,
+      final String serverName) {
+    ServerFilterForm serverFilterForm = new ServerFilterForm();
+    serverFilterForm.setApplicationIds(Collections.singletonList(applicationId));
+
+    Servers servers;
+    List<String> serverIds;
+
+    try {
+      serverFilterForm.setQ(URLEncoder.encode(serverName, "UTF-8"));
+      servers = sdk.getServersWithFilter(orgUuid, serverFilterForm);
+    } catch (IOException e) {
+      throw new GradleException("Unable to retrieve the servers.", e);
+    } catch (UnauthorizedException e) {
+      throw new GradleException("Unable to connect to Contrast.", e);
+    }
+
+    if (!servers.getServers().isEmpty()) {
+      serverIds = new ArrayList<String>();
+      for (Server server : servers.getServers()) {
+        serverIds.add(String.valueOf(server.getServerId()));
+      }
+    } else {
+      throw new GradleException(
+          "\n\nServer with name '"
+              + serverName
+              + "' not found. Make sure this server name appears in Contrast under the 'Servers' tab.\n");
+    }
+
+    return serverIds;
+  }
+
+  /** Retrieves appId from TeamServer using the OrgUuid and Appname */
+  private String getApplicationId(
+      final ContrastConfigurationExtension config, final String orgUuid) {
+    Applications applications;
+
+    if (!config.getAppName().isPresent()) {
+      throw new GradleException("AppName or Organization UUID not set");
+    }
+    try {
+      applications = sdk.getApplications(orgUuid);
+    } catch (IOException e) {
+      throw new GradleException("Unable to retrieve the applications: " + e);
+    } catch (UnauthorizedException e) {
+      throw new GradleException("Unable to connect to TeamServer: " + e);
+    }
+
+    for (Application application : applications.getApplications()) {
+      if (Objects.equals(config.getAppName().get(), application.getName())) {
+        return application.getId();
+      }
+    }
+
+    throw new GradleException("Application with name '${applicationName}' not found.");
+  }
+
+  private TraceFilterBody getTraceFilterBody(
+      final List<String> serverIds, final String minSeverity, final String computedAppVersion) {
+    TraceFilterBody body = new TraceFilterBody();
+    body.setSeverities(getSeverityList(minSeverity));
+    body.setAppVersionTags(Collections.singletonList(computedAppVersion));
+    if (serverIds != null) {
+      body.setServers(serverIds);
+    }
+    return body;
+  }
+
+  /**
+   * Creates a basic report for a Trace object
+   *
+   * @param trace Trace object
+   * @return String report
+   */
+  private String generateTraceReport(final Trace trace) {
+    return "Trace: "
+        + trace
+            .getTitle()
+            // TODO check regex
+            .replaceAll("\\{\\{\\#unlicensed\\}\\}", "(")
+            .replaceAll("\\{\\{\\/unlicensed\\}\\}", ")")
```

Comment:
No need for a regex here. If you're replacing the literal strings `{{#unlicensed}}` and {{/unlicensed}}`, use `String.replace`.